rpcs3/Emu/system_progress.cpp

```diff
@@ -82,7 +82,10 @@ void progress_dialog_server::operator()()
 			}
 		}
 
-		if (!skip_this_one && !native_dlg && (dlg = Emu.GetCallbacks().get_msg_dialog()))
+		if (!skip_this_one && !native_dlg && (Emu.BlockingCallFromMainThread([&dlg]()
```

Comment:
Just do this in get_msg_dialog. I don't understand why you do this in a super weird way here